nextcord/polls.py

```diff
@@ -0,0 +1,437 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from .emoji import Emoji
+from .enums import PollLayoutType, try_enum
+from .errors import InvalidArgument
+from .iterators import AnswerVotersIterator
+from .partial_emoji import PartialEmoji
+from .utils import MISSING
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .abc import Snowflake
+    from .message import Message
+    from .state import ConnectionState
+    from .types.emoji import PartialEmoji as PartialEmojiPayload
+    from .types.polls import (
+        Poll as PollData,
+        PollAnswer as PollAnswerPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        PollCreateRequest as PollCreateRequestPayload,
+        PollMedia as PollMediaPayload,
+        PollResults as PollResultsPayload,
+    )
+
+    # for reviewers: this typehint type is precedent in the nextcord codebase i.e.
+    # https://github.com/nextcord/nextcord/blob/master/nextcord/message.py#L68
+    EmojiInputType = Union[Emoji, PartialEmoji, str]
+
+
+def resolve_emoji(emoji: Union[Emoji, PartialEmoji]) -> PartialEmojiPayload:
+    if isinstance(emoji, Emoji):
+        return emoji._to_partial().to_dict()  # type: ignore
+
+    if isinstance(emoji, PartialEmoji):
+        return emoji.to_dict()  # type: ignore
```

Comment:
There is a `cast` for these things.
```suggestion
        return typing.cast(PartialEmojiPayload, emoji.to_dict())
```